df.announcements.xml

```diff
@@ -392,6 +393,55 @@
         <enum-item name='SATISFIED_MONARCH' since='v0.50.01'/>
         <enum-item name='MOUNTAINHOME' since='v0.50.01'/>
         <enum-item name='FOOD_WARNING' since='v0.50.01'/>
+
```

Comment:
I don't think we really need `num` because we have our own approach to enum metadata. And I don't think we really need the UNUSED values either, as long as they really are unused, if their purpose is just to pad out the enum.

Reply:
Are there going to be issues with `NONE`? `is_valid_enum_item` probably isn't sufficient to make sure we do attempt `df::global::d_init->announcements.flags[announcement_type::NONE]`.

Reply:
_every_ enum in toadycode has a NUM value, which we generally never include in structures since it's not actually a value the enum will ever have, and including it will actually break places where we use the values of an enum as an array index or as a . i've been including the `NONE` value in cases where it makes sense to do so, which is more often than not, especially as toady often uses `NONE` as an initializer for enum values

toady is a very idiosyncratic programmer, and while his practices seem to work for _him_ dfhack has a long tradition of having different programming practices, which is why we're not adopting some of toady's more, ahem, distinctive coding style choices

Reply:
> _every_ enum in toadycode has a `NUM` value, which we generally never include in structures since it's not actually a value the enum will ever have

I noticed `game_type` and `game_mode` seemed to have gained this for some reason.

Reply:
> I noticed `game_type` and `game_mode` seemed to have gained this for some reason. I assume it's to aid in disassembly or something?

that was done by ag in 2012, and they're not even correct: `NONE` should have value `-1` but in both of those enums, they don't.

i'd prefer that we fix old mistakes rather than expand on them.

Reply:
Should we create a PR to fix those, then?

Reply:
if you don't, i'll do it eventually when i get to canonicalization of those types. but this might well be weeks or even months, especially right now as i'm busy with pre-50.12